### PR TITLE
Send email - error journey

### PR DESCRIPTION
**Pull request by easternbloc**
Create a model for the email service
Add a email for the error endpoint
Tests

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -14,5 +14,15 @@ module.exports = {
...
+  },
+  email: {
+    safeMode: false,
+    port: 587,
+    host: 'email-smtp.eu-west-1.amazonaws.com',
+    auth: {
+      user: 'username',
+      pass: 'password'
+    },
+    from: 'brp@dsp.notprod.homeoffice.gov.uk'
```
Comment (JoeChapman) [inline config.js]: Why aren't these settings configurable from the environment?

Comment (vaijab) [inline config.js]: > Why aren't these settings configurable from the environment?

+1

Comment (easternbloc) [inline config.js]: You're not my dad.

File: controllers/check-details.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  // set template
+  switch (req.originalUrl) {
+    case '/permit-delivery/check-details':
+      model.set('template', 'permit');
+      break;
+    case '/permit-error/check-details':
+      model.set('template', 'error');
+      break;
+    case '/permit-lost-or-stolen/check-details':
```
Comment (JoeChapman) [inline controllers/check-details.js]: I'm not sure I like the way these urls are hard-coded in the controller. At the least I think the switch should partially match cases, but even better would be a centralised point that maintained the routes.

File: controllers/check-details.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      model.set('template', 'lost-or-stolen');
```
Comment (JoeChapman) [inline controllers/check-details.js]: The same could apply to the templates too.

File: controllers/check-details.js (hunk 2 above)
Comment (easternbloc) [inline controllers/check-details.js]: There's no need for it to partially match cases. I agree about there being a central location for this but I'm not quite sure what that should look like right now...

File: services/email/index.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+var transport = config.email.safeMode ?
+  require('nodemailer-stub-transport') : require('nodemailer-smtp-transport');
+
+function Emailer() {
+  this.transporter = nodemailer.createTransport(transport({
+    host: config.email.host,
+    port: config.email.port,
+    secure: false,
+    auth: config.email.auth,
+    debug: true,
```
Comment (JoeChapman) [inline services/email/index.js]: Perhaps `process.env.NODE_ENV === 'development' :? true : false`?

File: services/email/templates/error_html.mus [hunk 5]
```diff
@@ -0,0 +1,220 @@
+<!doctype html>
+<!--[if lt IE 7]>  <html class="ie ie6 lte9 lte8 lte7"> <![endif]-->
+<!--[if IE 7]>     <html class="ie ie7 lte9 lte8 lte7"> <![endif]-->
+<!--[if IE 8]>     <html class="ie ie8 lte9 lte8"> <![endif]-->
+<!--[if IE 9]>     <html class="ie ie9 lte9"> <![endif]-->
+<!--[if gt IE 9]>  <html> <![endif]-->
+<!--[if !IE]><!--> <html>             <!--<![endif]-->
+  <head>
+    <meta content="text/html;charset=utf-8" http-equiv="Content-Type">
+  <meta content="utf-8" http-equiv="encoding">
```
Comment (JoeChapman) [inline services/email/templates/error_html.mus]: no indentation

**Pull request merged**